talk/morelanguage/stl.tex

```diff
@@ -57,16 +56,63 @@
       \end{itemize}
     \item examples
       \begin{itemize}
-      \item string
-      \item tuple, list, forward\_list (\cpp11), vector, deque, array (\cpp11)
+      \item string, string\_view (\cpp17)
+      \item list, forward\_list (\cpp11), vector, deque, array (\cpp11)
       \item map, set, multimap, multiset
       \item unordered\_map (\cpp11), unordered\_set (\cpp11)
       \item stack, queue, priority\_queue
+      \item span (\cpp20)
       \end{itemize}
+    \item non-containers: pair, tuple (\cpp11), optional (\cpp17), variant (\cpp17), any (\cpp17)
+    \item see also the \href{https://en.cppreference.com/w/cpp/string/basic_string}{string} and \href{https://en.cppreference.com/w/cpp/container}{container library} on cppreference
     \end{itemize}
   \end{block}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[11]{Containers: std::vector}
+  \begin{cppcode*}{}
+    #include <vector>
+    std::vector<T> v{5, 3, 4}; // 3 Ts, 5, 3, 4
+    std::vector<T> v(100);     // 100 default constr. Ts
+    std::vector<T> v(100, 42); // 100 Ts with value 42
```

Comment:
Now we would have to explain further the notion of initializer list and why we were not really complete when saying "just use curly braces for calling constructors earlier in the course. I think we should indeed do that, but maybe in the expert course. I'll open an issue on this

Reply:
resolved in issue #155